dashboard/app/helpers/application_helper.rb

```diff
@@ -222,4 +222,30 @@ def tracking_pixel_enabled
   def page_mode
     PageMode.get(request)
   end
+
+  def response_for_share_failure(share_failure)
+    return nil unless share_failure
+    {}.tap do |failure|
+      failure[:message] = share_failure_message(share_failure.type)
+      failure[:type] = share_failure.type
+      failure[:contents] = share_failure.content unless share_failure.type == ShareFiltering::FailureType::PROFANITY
+    end
+  end
+
+  private
+
+  def share_failure_message(failure_type)
+    case failure_type
+      when ShareFiltering::FailureType::EMAIL
+        t('share_code.email_not_allowed')
+      when ShareFiltering::FailureType::ADDRESS
+        t('share_code.address_not_allowed')
+      when ShareFiltering::FailureType::PHONE
+        t('share_code.phone_number_not_allowed')
+      when ShareFiltering::FailureType::PROFANITY
+        t('share_code.profanity_not_allowed')
+      else
+        raise ArgumentError.new("Unknown share failure type #{failure_type}")
+    end
```

Comment:
nit: The `case`, `when`, `else`, and `end` should have the same indentation.
Not yet enforced, though it will be soon.
https://github.com/bbatsov/ruby-style-guide#indent-when-to-case